### DISCUSSION TITLE
Add roles, role permissions and email templates at config level

### DISCUSSION THREAD

**Discussion by tiberiumihai** [strapi/strapi#5948] [Legacy Topics]
Is there any way I can add my own roles, role permissions, and email templates configured via config files?

I've tried using ```extensions/user-permissions/functions/bootstrap.js``` but I don't see any change in my email templates.

I've also stumbled upon this: [https://strapi.io/documentation/3.0.0-alpha.x/configurations/configurations.html#get-settings](https://strapi.io/documentation/3.0.0-alpha.x/configurations/configurations.html#get-settings)

But I have no idea where that goes. Should be added in a ```bootstrap.js``` file or should I create a plugin that does all that?

If we can configure these things at config/init level, then the migration and scaling our strapi apps in a cluster would be easier.

Thanks!

**Comment by MattieBelt** [comment]
@tiberiumihai They are working on a new config, check out #5747 

> This PR is a MVC (minimum viable change) to enhance user usage without achieving the full refactoring we will be doing for the next major release.

**Comment by lauriejim** [comment]
Hello! If you want to programmatically create / update roles and permissions I suggest you check how it works.
To do that, you will have to check the format of the object that is send when you create / update a permission from the admin (browser network inspector)
That will call [UserPermission controller](https://github.com/strapi/strapi/blob/0c6d39297f6f8a4f983e22fb48256b42da2a8605/packages/strapi-plugin-users-permissions/controllers/UsersPermissions.js) that mainly call service's functions.
So if you follow the logic, you will be able to use programmatically the UserPermission service, with the correct param format (check it in your browser network) to create new roles.

**Reply by tiberiumihai** [reply]
Thank you very much! 

I'll add my own permissions in bootstrap.js at startup.
1 - First check the roles aren't there;
2 - If they are not there, then add them all.
